python/cudf/udf_cpp/strings/src/strings/udf/udf_apis.cu

```diff
@@ -98,6 +129,24 @@ void free_udf_string_array(cudf::strings::udf::udf_string* d_strings,
                      [d_strings] __device__(auto idx) { d_strings[idx].clear(); });
 }
 
+/**
+ * @copydoc free_managed_udf_string_array
+ *
+ * @param stream CUDA stream used for allocating/copying device memory and launching kernels
+ */
+void free_managed_udf_string_array(cudf::strings::udf::managed_udf_string* managed_strings,
+                                   cudf::size_type size,
+                                   rmm::cuda_stream_view stream)
+{
+  thrust::for_each_n(rmm::exec_policy(stream),
+                     thrust::make_counting_iterator(0),
+                     size,
+                     [managed_strings] __device__(auto idx) {
+                       managed_strings[idx].udf_str.clear();
```

Comment:
Should this be replaced with a call to the destructor, similarly to https://github.com/rapidsai/cudf/pull/13060/files#r1158371391 ?

Reply:
This follows the pattern from [the other API](https://github.com/rapidsai/cudf/blob/branch-23.06/python/cudf/udf_cpp/strings/src/strings/udf/udf_apis.cu#L98). But this is a good point nonetheless. 

@davidwendt should we be calling the full destructor in _both_ these cases?

Reply:
From my understanding, the lifetime of the managed-udf-string is managed by the meminfo object so I don't think  you need this call at all, right? The meminfo's manage-udf-string will be cleaned up Numba?

Reply:
that is true except in the case where we want to return the string outside of the function, meaning the result of the full UDF computation is a new string. In that case we make sure the strings are _not_ cleaned up and still exist to be transformed into the final column.

Technically we do this by adding an `incref` to every string that gets placed into the output array, so at the end of the kernel the `refct` associated with those strings is always `1` still.

Reply:
You just don't want to be calling the destructor twice. If you were to call the destructor here and then meminfo calls it again later when `refct==0` for the same object that would be bad. Are you also destroying the meminfo objects after they get converted to a strings column?

Reply:
Oh wait, I see that you are destroying the `meminfo` right below this line! So yes, then calling the destructor here makes sense.